### ISSUE TITLE
options.jsonify is always forced to true

### ISSUE THREAD

**Issue by Venryx** [agilgur5/mst-persist#17]
The code in `index.js` has a bug: https://github.com/agilgur5/mst-persist/blob/master/src/index.ts

Look closely at the highlighted usages:
![](https://i.imgur.com/Q9WzbQj.png)

Note this line:
```
if (!jsonify) { jsonify = true } // default to true like mobx-persist
```

And note that there is no line after that which sets the jsonify variable's value.

Basically, the line above forces jsonify to *always* be `true`, even if the user explicitly sets it to `false`.

You probably meant this instead:
```
if (jsonify == null) { jsonify = true } // default to true like mobx-persist
```

**Comment by agilgur5** [comment]
Ah, poop, well that's a dumb mistake :sweat_smile: . Thanks for catching this and filing the issue @Venryx !

> You probably meant this instead:
> 
> ```
> if (jsonify == null) { jsonify = true } // default to true like mobx-persist
> ```

Actually, I meant `undefined`, since if `jsonify` is not specified, it'll be `undefined`, not `null`. So that would look like:

```typescript
if (typeof jsonify === 'undefined') { jsonify = true } // default to true like mobx-persist
```

At some point I just had `jsonify = true` as a default arg in the destructuring, and then once I wrote the somewhat complex (and later even more in #15) defaults logic for the `storage` config, I thought it would make sense to do all defaults outside of destructuring. That clearly invited some bugs 😖 . 

`=== undefined` is meant to replicate how default args work, but I think I'll just make it a default arg instead to fix as I'm already using one [for Transforms](https://github.com/agilgur5/mst-persist/pulls/16) anyway and, as I found out during its implementation, default args make for better typings too.
